src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -63,33 +63,30 @@ parseTypeSynonymDeclaration =
                          <*> many (indented *> kindedIdent)
                          <*> (indented *> equals *> noWildcards parsePolyType)
 
-parseValueWithWhereClause :: TokenParser Expr
-parseValueWithWhereClause = do
-  indented
-  value <- parseValue
-  whereClause <- P.optionMaybe $ do
-    indented
-    reserved "where"
-    indented
-    mark $ P.many1 (same *> parseLocalDeclaration)
-  return $ maybe value (`Let` value) whereClause
-
 parseValueDeclaration :: TokenParser Declaration
 parseValueDeclaration = do
-  name <- parseIdent
   binders <- P.many parseBinderNoParens
-  value <- Left <$> (indented *>
-                       P.many1 ((,) <$> parseGuard
-                                    <*> (indented *> equals *> parseValueWithWhereClause)
-                               ))
-       <|> Right <$> (indented *> equals *> parseValueWithWhereClause)
-  return $ ValueDeclaration name Public binders value
-
-parseBoundValueDeclaration :: TokenParser Declaration
-parseBoundValueDeclaration = do
-  binder <- parseBinderLocalDecl
-  value <- indented *> equals *> parseValueWithWhereClause
-  return $ BoundValueDeclaration binder value
+  case binders of
+    VarBinder name : binders' -> do
+      value <- Left <$> (indented *>
+                          P.many1 ((,) <$> parseGuard
+                                        <*> (indented *> equals *> parseValueWithWhereClause)
+                                  ))
+          <|> Right <$> (indented *> equals *> parseValueWithWhereClause)
+      return $ ValueDeclaration name Public binders' value
+    [binder] -> BoundValueDeclaration binder <$> (indented *> equals *> parseValueWithWhereClause)
+    _ -> P.unexpected "unexpected binders in value declaration"
```

Comment:
Parsec will print out "unexpected unexpected ..." here. Can you try it and see what the error looks like?